### PR TITLE
Export field conf to yaml file

### PR DESCRIPTION
**Pull request by stonebuzz**
Export fields container configuration to YAML file 

ex 

```yaml
container:
    2-Peripheral:
        id: 2
        name: ville
        itemtype: Peripheral
        fields:
            -
                id: 11
                label: simpletext
                xml_node: simpletextfield
                type: text
                ranking: '1'
                default_value: ''
                mandatory: '0'
                possible_value: ''
            -
                id: 12
                label: yesno
                xml_node: yesnofieldtwo
                type: yesno
                ranking: '2'
                default_value: ''
                mandatory: '0'
                possible_value:
                    -
                        id: 0
                        value: 'No'
                    -
                        id: 1
                        value: 'Yes'
```

export available for all containers

image

 or just one

image

Signed-off-by: Stanislas <skita@teclib.com>

### CODE REVIEW

File: setup.php [hunk 1]
```diff
@@ -239,3 +244,115 @@ function plugin_fields_checkFiles($force = false) {
...
+
+            foreach ($itemtypes as $itemtype) {
+               $fields_obj = new PluginFieldsField();
+               // to get translation
+               $container["itemtype"] = PluginFieldsContainer::getType();
+               $yaml_conf['container'][$container['id']."-".$itemtype]["id"]     = (int) $container['id'];
+               $yaml_conf['container'][$container['id']."-".$itemtype]["name"]   = PluginFieldsLabelTranslation::getLabelFor($container);
+               $yaml_conf['container'][$container['id']."-".$itemtype]["itemtype"] = $itemtype;
+               $yaml_conf['container'][$container['id']."-".$itemtype]["type"] = $container['type'];
+               $yaml_conf['container'][$container['id']."-".$itemtype]["subtype"] = $container['subtype'];
```
Comment (cedric-anne) [inline setup.php:281]: I found this format more readable (and I added an empty "fields" element to ensure it exists even if container does not contain fields).
```suggestion
               $yaml_conf['container'][$container['id']."-".$itemtype] = [
                  "id" => (int) $container['id'],
                  "name" => PluginFieldsLabelTranslation::getLabelFor($container),
                  "itemtype" => $itemtype,
                  "type" => $container['type'],
                  "subtype" => $container['subtype'],
                  "fields" => [],
               ];
```

File: setup.php [hunk 2]
```diff
@@ -239,3 +244,115 @@ function plugin_fields_checkFiles($force = false) {
...
+               $fields = $fields_obj->find(["plugin_fields_containers_id" => $container['id']]);
+               if (count($fields)) {
+                  foreach ($fields as $field) {
+                     $tmp_field= [];
+                     if ($field['is_active'] && !$field['is_readonly']) {
```
Comment (cedric-anne) [inline setup.php:286]: Maybe this could be directly filtered in the SQL query.

File: setup.php [hunk 3]
```diff
@@ -239,3 +244,115 @@ function plugin_fields_checkFiles($force = false) {
...
+/**
+ * Check configuration process
+ *
+ * @param boolean $verbose Whether to display message on failure. Defaults to false
+ *
+ * @return boolean
+ */
+function plugin_fields_check_config($verbose = false) {
+   return true;
+}
```
Comment (cedric-anne) [inline setup.php:358]: This method is not mandatory anymore in GLPI 9.5.
```suggestion
```

File: setup.php [hunk 4]
```diff
@@ -72,8 +74,11 @@ function plugin_init_fields() {
 
    $PLUGIN_HOOKS['csrf_compliant']['fields'] = true;
 
-   // manage autoload of plugin custom classes
+   // anage autoload of plugin
    include_once(PLUGINFIELDS_DIR . "/inc/autoload.php");
+
+   // manage autoload of plugin custom classes
+   include_once(PLUGINFIELDS_DIR . "/vendor/autoload.php");
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply setup.php]: done

File: setup.php [hunk 5]
```diff
@@ -239,3 +244,115 @@ function plugin_fields_checkFiles($force = false) {
       }
    }
 }
+
+function plugin_fields_exportBlockAsYaml($container_id = null) {
+   global $DB;
+
+   $plugin = new Plugin();
+   $yaml_conf = [];
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply setup.php]: done

File: setup.php (hunk 1 above)
Comment (stonebuzz) [reply setup.php]: done

File: setup.php (hunk 2 above)
Comment (stonebuzz) [reply setup.php]: done

File: setup.php (hunk 3 above)
Comment (stonebuzz) [reply setup.php]: done

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**